### ISSUE TITLE
API: Align  User permissions for View / Add / Update / Delete entity

### ISSUE THREAD

**Issue by Schnitzel** [uselagoon/lagoon#154]
currently as a authenticates user I can only read API Objects, but I would also like to create and update Objects.

Here the following permissions that should be necessary:

**User with Customer Access (aka User's ssh key is added to a customer)**
- Projects: View, Add (only to customers, I have access to), Update & Delete
- Customers: View (cannot update, create or delete)
- Environments: View (cannot update, create or delete)
- OpenShift: View (cannot update, create or delete)
- SSH Keys: View, Update, Delete (all the ones that are attached to a customer or project the user has access to); Add 
- SSHKeysToProject: View, Update, Delete (all the ones that are attached to a project the user has access to); Add (Can choose all keys, as the key might just has been created and is not attached to anything yet, so we need to be able to choose all ssh keys)
- SSHKeysToCustomer: View, Update, Delete (all the ones that are attached to a customer the user has access to); Add (Can choose all keys, as the key might just has been created and is not attached to anything yet, so we need to be able to choose all ssh keys)
- NotificationSlack: View, Update, Delete (all the ones that are attached to a project the user has access to); Add 
- NotificationToProject: View, Update, Delete (all the ones that are attached to a project the user has access to); Add (Can choose notifications, as the key might just has been created and is not attached to anything yet, so we need to be able to choose all notifications)

**User with Project Access (aka User's ssh key is added to a Project)**
Basically the same as Customer Access just with the difference that there is no access to the Customer Object and SSHKeysToCustomer at all.

Needs review by @ryyppy

**Comment by Schnitzel** [comment]
also note that we will actually implement additional Roles:
https://github.com/amazeeio/lagoon/issues/156
But these roles are not due for V1.0. So I'm okay if we first implement it in a way that all Users have edit access.

**Comment by ryyppy** [comment]
To make things easier, this is the checklist to verify implemented functionality for **User permissions**:

**Projects:**
- [x] View
- [x] Add (only to customer, I have access to)
- [x] Update (project-id which the user has credentials for)
- [x] Delete

**Customers:**
- [x] View (cannot update, create or delete)

**Environments:**
- [x] View (cannot update, create or delete)

**Openshift:**
- [x] View (cannot update, create or delete)

**SshKeys:**
- [x] View all SshKeys (assigned & unassigned) (the ones the user has access to through customer & project permissions)
- [x] View **unassigned** SshKeys (those which are created, but not yet attached to a project / customer)
- [x] Add
- [x] Update (all the ones that are attached to a customer or project the user has access to + all unassigned keys)
- [x] Delete (all the ones that are attached to a customer or project the user has access to + all unassigned keys)

**SshKeysToProject:**
- [x] View
- [x] Add (Can choose all keys, as the key might just has been created and is not attached to anything yet, so we need to be able to choose all ssh keys)
- [x] ~~Update (all the ones that are attached to a project the user has access to)~~
- [x] Delete (all the ones that are attached to a project the user has access to)

**SshKeysToCustomer:**
- [x] View
- [x] Add (Can choose all keys, as the key might just has been created and is not attached to anything yet, so we need to be able to choose all ssh keys)
- [x] ~~Update (all the ones that are attached to a customer the user has access to)~~
- [x] Delete (all the ones that are attached to a customer the user has access to)

**NotificationSlack:**
- [x] View
- [x] Add
- [x] Update (all the ones that are attached to a project the user has access to)
- [x] Delete (all the ones that are attached to a project the user has access to)

**NotificationRocketChat:**
- [x] View
- [x] Add
- [x] Update (all the ones that are attached to a project the user has access to)
- [x] Delete (all the ones that are attached to a project the user has access to)

**NotificationToProject:**
- [x] ~~View~~
- [x] Add (Can choose notifications, as the key might just has been created and is not attached to anything yet, so we need to be able to choose all notifications)
- [x] Update (all the ones that are attached to a project the user has access to)
- [x] Delete (all the ones that are attached to a project the user has access to)
